pkg/kubernetes/adjust.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "base1/patterns",
+    "base1/mustache",
+    "kubernetes/client"
+], function(jQuery, cockpit, patterns, Mustache, kubernetes) {
+    "use strict";
+
+    var _ = cockpit.gettext;
+    var $ = jQuery.scoped("#adjust-dialog", patterns);
+
+    var kube = null;
+
+    var dialog = $("#adjust-dialog", document);
+
+    dialog
+        .on("show.bs.modal", function(ev) {
+            kube = kubernetes.k8client();
+
+            var uid = $(ev.relatedTarget).attr("data-uid");
+            var service = kube.objects[uid];
+
+            if (!service) {
+                console.warn("no such service:", uid);
+                dialog.modal("hide");
+                return;
+            }
+
+            var meta = service.metadata || { };
+            var spec = service.spec || { };
+
+            /* We can't edit the name for now */
+            $("#adjust-name")
+                .val(meta.name)
+                .attr("disabled", "disabled");
+
+            /* Generate table rows for each replication controller this service matches */
+            var rcs = kube.select(spec.selector || { }, meta.namespace, "ReplicationController");
```

Comment:
Instead of selecting replication controllers by label, we must select them by the labels in their template.